### PR TITLE
feat: add `system.queries `system table

### PR DESCRIPTION
**Pull request by alamb**
# Rationale

re https://github.com/influxdata/conductor/issues/795

I want to get a snapshot of actual customer queries that were run against an IOx system so that with a snapshot of the data I can reproduce, profile, and optimize IOx's query performance

To do so I need to know what actual queries were sent to IOx, in a form I can replay them. 

# Changes:
1. Record the last 100 (picked arbitrarily) queries (both SQL and gRPC) queries that are run in memory
2. Expose that information via `system.queries` system table
3. Tests

# Design notes
I put the log on `Db` (as it is a db-wide query log), which needed a little code rearranging in the storage service. It seems a little less than ideal but I don't think it is terrible either and I can't come up with anything better

### CODE REVIEW

**Comment by tustvold** [comment]
FYI - https://github.com/influxdata/influxdb_iox/pull/3334 should fix the pbjson

File: influxdb_iox/src/influxdb_ioxd/server_type/database/http.rs [hunk 1]
```diff
@@ -252,6 +252,8 @@ async fn query(
 
     let db = server.db(&db_name)?;
 
+    db.record_query("sql", &q);
```
Comment (alamb) [inline influxdb_iox/src/influxdb_ioxd/server_type/database/http.rs:255]: here is where SQL queries (that come from HTTP) are recorded

File: influxdb_iox/src/influxdb_ioxd/server_type/database/rpc/flight.rs [hunk 2]
```diff
@@ -170,6 +171,8 @@ impl Flight for FlightService {
             .db(&database)
             .map_err(default_server_error_handler)?;
 
+        db.record_query("sql", &read_info.sql_query);
```
Comment (alamb) [inline influxdb_iox/src/influxdb_ioxd/server_type/database/rpc/flight.rs:174]: SQL queries from flight are recorded here

File: query_tests/cases/in/all_chunks_dropped.expected [hunk 3]
```diff
@@ -9,6 +9,7 @@
 | public        | system             | chunk_columns       | BASE TABLE |
 | public        | system             | operations          | BASE TABLE |
 | public        | system             | persistence_windows | BASE TABLE |
```
Comment (alamb) [inline query_tests/cases/in/all_chunks_dropped.expected:11]: this is a strange test -- I will update it to not be a replica of system tables -- will change it so it only selects iox tables (not system tables)

**Review by e-dard — APPROVED** [review]
I looks through this and it all makes sense to me!

File: arrow_util/src/test_util.rs [hunk 4]
```diff
@@ -89,3 +92,39 @@ pub fn sort_record_batch(batch: RecordBatch) -> RecordBatch {
 
     RecordBatch::try_new(batch.schema(), sort_output).unwrap()
 }
+
+/// Apply a
```
[the remark this answers is not in the document]
Comment (alamb) [reply arrow_util/src/test_util.rs:96]: hahah -- will fix that

[the remark this answers is not in the document]
Comment (alamb) [reply arrow_util/src/test_util.rs]: In https://github.com/influxdata/influxdb_iox/pull/3328/commits/91139468eaf0a915a47867eb41f1a32ee4c5073a

**Pull request merged**